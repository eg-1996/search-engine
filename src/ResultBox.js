import React,{ useState } from 'react';
import { List, Grid } from '@material-ui/core';
import UserItem from './UserItem';
import UserCard from './UserCard';
import { getUserProfile } from './Api';

const ResultBox = ({ results, setErrorMessage }) => {
    const [selectedUser, setSelectedUser] = useState(null);
    const handleClick = (user) => { 
        getUserProfile(user.url, setSelectedUser, setErrorMessage);
    }

    const userList =[];
    results.items.forEach(user => {
        userList.push(<UserItem user={user} handleClick={handleClick} key={user.id}/>)
    });

    return (<Grid container>
        <Grid item xs={6}>
            <List>
                {userList}
            </List>
        </Grid>
        <Grid item xs={6}>
            <UserCard user={selectedUser}/>
        </Grid>
        
    </Grid>)
};

ResultBox.defaultProps = {
    results:{
        items: [],
        total_count: 0
     }
}
export default ResultBox;