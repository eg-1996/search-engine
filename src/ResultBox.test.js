import React from 'react';
import renderer from 'react-test-renderer';
import ResultBox from './ResultBox';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <ResultBox />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});