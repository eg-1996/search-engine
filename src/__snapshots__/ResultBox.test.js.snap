// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  >
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
  />
</div>
`;
